Tests/kaas/k8s-version-policy/k8s_version_policy_test.py

```diff
@@ -0,0 +1,181 @@
+"""
+Pytest based unit tests for k8s_version_policy.
+
+(c) Martin Morgenstern <martin.morgenstern@cloudandheat.com>, 3/2024
+SPDX-License-Identifier: CC-BY-SA-4.0
+"""
+
+from datetime import datetime
+from pathlib import Path
+from unittest import mock
+import json
+import logging
+
+import pytest
+
+from k8s_version_policy import (
+    check_k8s_version_recency,
+    is_high_severity,
+    parse_branch_info,
+    parse_cve_version_information,
+    parse_github_release_data,
+    parse_version,
+    K8sBranch,
+    K8sBranchInfo,
+    K8sVersion,
+    VersionRange,
+)
+
+
+HERE = Path(__file__).parent
+
+
+@pytest.fixture
+def release_data():
+    """Fixture that loads our releases.json test data."""
+    with open(Path(HERE, "testdata", "releases.json")) as stream:
+        return json.load(stream)
+
+
+# The test simulates checking a v1.28.5 cluster at different
+# points in time. The version becomes "unrecent" a week after
+# the successor v1.28.6 is released.
+# In our test data set, v1.28.6 was released on 2024-01-17.
+K8S_VERSION = K8sVersion(1, 28, 5)
+EXPECTED_RECENCIES = {
+    datetime(2024, 1, 17): True,
+    datetime(2024, 1, 24): True,
+    datetime(2024, 1, 25): False,
+}
+
+
+@pytest.mark.parametrize("ref_time, expected_recent", EXPECTED_RECENCIES.items())
+def test_check_version_recency_without_cve(release_data, ref_time, expected_recent):
+    with mock.patch("k8s_version_policy.datetime", wraps=datetime) as dt:
```

Comment:
I know that I'm being obnoxious right now, but should we just make the `check_k8s_version_recency` API more general? It could have an optional argument `check_date=None` (and if `None` is supplied, the function will use `datetime.now`).

Reply:
Yeah, let's try to avoid monkey patching were possible. This `mock.patch(…)` call is also error-prone, because it relies so much on internal details of the module under test.

Reply:
Ahh, unfortunately we would need to carry down this through multiple functions or methods. Without a refactoring this will look ugly, so I would leave it as-is for the time being.